### PR TITLE
Refactored :class:`~.BarChart` and made it inherit from :class:`~.Axes`.

### PR DESCRIPTION
**Pull request by hydrobeam**
## Overview: What does this pull request change?

- :class:`~.BarChart` now inherits from :class:`~.Axes`, allowing it to use `Axes`' methods. Also improves `BarChart`'s configuration and ease of use.
- Added :meth:`~.BarChart.add_bar_labels` to annotate the value of each bar of a :class:`~.BarChart`.

## Motivation and Explanation: Why and how do your changes improve the library?

This PR is a continuation of the PR started by @LedMeExplain in #1916. The PR was inactive for a while, so I decided to implement the suggestions I had left, clean up the code + add documentation. See #1916 for a more full description of the changes, but here are the changes that I made when taking it over:

- removed `get_bar_labels` method in favour of just using built in `add_labels` for a `NumberLine`
- Removed uncessary parameters that could just be initialized via `axis_config/x_axis_config/y_axis_config`.
    - `y_label_buff`
    - `x_label_scale_value`
    - `y_number_scale`
    - `x_label_buff`
    - `y_include_numbers`.
  
- Removed `y_step` and merged its capabilities with `y_range` 
- Renamed `bar_buff` to `bar_width`. (same functionality, just that bigger number = bigger bar, not smaller bar)
- Removed getter methods in favour of directly accessing attributes via the axes.
- Adjusted `get_bar_labels` (returns mobs) into `add_bar_labels` (just adds them).
 

## Links to added or changed documentation pages

https://manimce--2387.org.readthedocs.build/en/2387/reference/manim.mobject.probability.BarChart.html
## Further Information and Comments

I'll add some tests soon if there's no objection to the changes in this PR.

Also, this is technically a breaking change, not sure if just breaking it is fine or if we should go into some deprecation period. (I favour just breaking it since `BarChart` isn't too commonly used).

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by hydrobeam** [comment]
Managed to fix it with the latest commit.

**Review by icedcoffeeee — CHANGES_REQUESTED** [review]
Two concerns:
1. The previous implementation had `.change_bar_values`, maybe add it here too?
2. Can you add tests? It turns out it hasn't been tested before.

Other than that, everything else LGTM

**Comment by hydrobeam** [comment]
I've written tests (adapted from @LedMeExplain's showcase in the original PR) and fixed negative bar values + rewrote some examples.

**Comment by hydrobeam** [comment]
Merged #2397  into this PR to remove the need for the `x_label_constructor` parameter.

**Review by Darylgolden — COMMENTED** [review]
I think `BarChart` is better placed on `coordinate_systems.py` rather than `probability.py`. What do you think?

**Comment by GameDungeon** [comment]
Honestly, why don't we just delete probability.py?

**Comment by Darylgolden** [comment]
> Honestly, why don't we just delete probability.py?

That was what I was getting at.

**Comment by hydrobeam** [comment]
> I think BarChart is better placed on coordinate_systems.py rather than probability.py. What do you think?

I disagree, it's not an essential component of graphing and is more of an auxillary class for displaying data.

> Honestly, why don't we just delete probability.py?

I think it's reasonable to have at least one supported way of showing data in Manim. Before this PR the class was essentially an unusable mess, but with these changes it's much more sensible and belongs in the library. As for `SampleSpace`, I'm indifferent. It has a poor implementation and is not nearly as ubiquitous as a bar chart.

**Review by behackl — APPROVED** [review]
This seems like useful improvements. I'm fine with merging this.

**Review by icedcoffeeee — APPROVED** [review]
LGTM, thank you for the improvements :)

File: manim/mobject/probability.py [hunk 1]
```diff
@@ -220,119 +220,258 @@ class BarChart(VGroup):
...
+        self.add_to_back(self.bars)
+
+    def get_bar_labels(
+        self,
+        color: Optional[Color] = None,
+        font_size: float = 24,
+        buff: float = MED_SMALL_BUFF,
+        label_constructor: "VMobject" = Tex,
+    ):
+        """Annotates each bar with its corresponding value. Use `self.bar_labels` to access the
```
Comment (hydrobeam) [inline manim/mobject/probability.py:369]:
```suggestion
        """Annotates each bar with its corresponding value. Use ``self.bar_labels`` to access the
```

**Pull request merged**